core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -929,6 +929,13 @@ private[spark] class TaskSetManager(
             info.id, taskSet.id, tid, ef.description))
           return
         }
+        if (ef.className == classOf[SparkUserException].getName) {
```

Comment:
I wonder whether this logic can be generic and apply to all exceptions that extend `SparkThrowable` and have an error class defined?

cc @cloud-fan @gengliangwang @HyukjinKwon

Reply:
+1, we should skip retry if the exception is `SparkThrowable` and the error class is present.

Reply:
One thing we should think about is how to differentiate user-facing error and user-triggered error. We may still need to retry for user-facing error, e.g. file read error which can be transient.

One idea is to have a special prefix for error classes that should still trigger retry, such as file read error and OOM, which shouldn't be many.

Reply:
Or we can create a base trait for transient errors.

Reply:
Thank you very much for review and sorry for the late response. I will try to modify the code according  to your suggestions @aokolnychyi @cloud-fan